z3/snap.py

```diff
@@ -583,8 +606,60 @@ def main():
         sys.stderr.write("Configuration error! {} is not set.\n".format(err))
         sys.exit(1)
 
-    bucket = boto.connect_s3(s3_key_id, s3_secret, **extra_config).get_bucket(bucket)
-
+    s3 = boto.connect_s3(s3_key_id, s3_secret, **extra_config)
+    try:
+        bucket = s3.get_bucket(bucket_name)
+    except boto.exception.S3ResponseError as e:
+        if e.error_code == 'NoSuchBucket':
+            # Let's try creating it
+            bucket = s3.create_bucket(bucket_name)
+            print("Created bucket {}: {}".format(bucket_name, bucket), file=sys.stderr)
+        else:
+            raise
+        
+    try:
+        lifecycle = bucket.get_lifecycle_config()
+    except boto.exception.S3ResponseError:
+        lifecycle = None
+
+    # See if we have a lifecycle rule for glacier.
+    # The rule name will depend on the S3_PREFIX -- if
+    # that's empty, then the rule is just "z3 transition";
+    # otherise, it is "z3 transition ${S3_PREFIX}" (but with
+    # '/' converted to ' ').
+    lifecycle_rule_name = "z3 transition"
+    if args.s3_prefix:
+        lifecycle_rule_name += " " + args.s3_prefix.replace("/", " ")
+    while lifecycle_rule_name.endswith(" "):
+        lifecycle_rule_name = lifecycle_rule_name[:-1]
+        
+    rule_index = None
+    for indx, rule in enumerate(lifecycle or []):
+        if rule.id == lifecycle_rule_name:
+            rule_index = indx
+            break
+        
+    if not args.use_glacier:
+        # If we don't use glacier, we want to remove the lifecycle policy
+        # if it exists
+        if rule_index is not None:
+            lifecycle.pop(rule_index)
+    else:
+        if rule_index is None:
+            # Okay, we need to add a lifecycle
+            if lifecycle is None:
+                lifecycle = boto.s3.lifecycle.Lifecycle()
+            transition=boto.s3.lifecycle.Transition(days=1, storage_class="GLACIER")
+            print("trasition rule = {}".format(transition), file=sys.stderr)
+            print("prefix = {}".format(args.s3_prefix or None), file=sys.stderr)
+            lifecycle.add_rule(id=lifecycle_rule_name,
+                               status="Enabled",
+                               prefix=args.s3_prefix or None,
+                               transition=transition)
+            print("lifecycle = {}".format(lifecycle.to_xml()), file=sys.stderr)
+    if lifecycle is not None:
+        bucket.configure_lifecycle(lifecycle)
```

Comment:
One of the reasons I did that was that I have it set up to remove the lifecycle rule if use_glacier is no longer in the configuration.

Changed it.